server/server.go

```diff
@@ -118,6 +120,19 @@ type ArgoCDServerOpts struct {
 	RepoClientset   reposerver.Clientset
 }
 
+// initializeDefaultProject creates the default project if it does not already exist
+func initializeDefaultProject(opts ArgoCDServerOpts) error {
+	defaultProj := &v1alpha1.AppProject{
+		ObjectMeta: metav1.ObjectMeta{Name: common.DefaultAppProjectName, Namespace: opts.Namespace},
+		Spec:       v1alpha1.AppProjectSpec{},
+	}
+	_, err := opts.AppClientset.ArgoprojV1alpha1().AppProjects(opts.Namespace).Get(common.DefaultAppProjectName, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		_, err = opts.AppClientset.ArgoprojV1alpha1().AppProjects(opts.Namespace).Create(defaultProj)
+	}
```

Comment:
I prefer the style of just blindly calling Create() and ignoring AlreadyExists error. This makes the max total API calls to 1, and if we ever go to multiple replicas for the API server, eliminates the race.

Reply:
Great point! I'll make that change!